### ISSUE TITLE
ECS Service Discovery not respecting TTL when updating service

### ISSUE THREAD

**Issue by matthewduren** [aws/amazon-ecs-agent#1594]
### Summary
When updating a service or otherwise scaling out ECS tasks for a service that uses Service Discovery, tasks are being stopped before reaching the TTL of the service discovery record(s).

### Description
When updating a service or otherwise scaling out ECS tasks for a service that uses Service Discovery, tasks are being stopped before reaching the TTL of the service discovery record(s).

To reproduce - create an ECS service from a simple "hello world" type task definition that runs forever and does nothing.  Set min healthy to 100, max to 200, count to 1.  Setup service discovery and create a DNS record with a long TTL, say 300s.  Update the service to use a new revision of the task definition (no changes to the task def needed), and note that the tasks are stopped before the TTL time is reached.

### Expected Behavior
ECS Agent should remove the route53 record(s), and then wait to stop the tasks after the TTL duration has elapsed.

### Observed Behavior
ECS Agent does not wait any additional time when stopping tasks for services that use service discovery.

**Comment by matthewduren** [comment]
Right, when it starts draining it should immediately remove the r53 record. After the TTL has elapsed the normal SIGTERM signal should be sent to the container, followed by SIGKILL 30 seconds later if the task is still up just like how tasks that don't use service discovery behave.

**Comment by melbourne2991** [comment]
Was hoping to avoid using an lb and this came to mind, sad to see it's an unresolved issue :(

**Comment by nikushi** [comment]
I'm on the same issue. 

I am building an infrastructure for gRPC services by using ECS Fargate and it's service discovery feature, without having ELB. The communication between services is transferred by Envoy proxy. Each envoy listener is served via ECS' service discovery. 

I got gRPC unavailable errors during updating a service. The envoy that is transferring a request to other envoy could lose all upstream connections since there is a moment that the envoy only knows old container's ip addresses, which are already dead by SIGTERM. As workaround, I configure the DNS TTL as very short value such as `3s`, though I got still errors in a short period of time(about in 10 seconds).

I hope that the issue will be resolved.

**Comment by kuongknight** [comment]
I hope that the issue will be resolved too :(

**Comment by afawaz2** [comment]
+1.
